packages/rule-meta-charset-utf-8/src/rule.ts

```diff
@@ -143,6 +164,7 @@ export default class MetaCharsetUTF8Rule implements IRule {
              */
         };
 
+        context.on('fetch::end::html', setReceivedDom);
```

Comment:
Why two events? if you already have the html in `fetch::end::html`, why don't validate there?

Reply:
@sarvaje See https://github.com/sonarwhal/sonarwhal/pull/900/files#diff-427f2e06ab7b36990c5b6ddb1f0b1f85R56.

Reply:
I also tried that at the beginning but because the elements are different I'd have to change quite a few things from `context.repo` and I wasn't sure it was worth it.

I could look into doing everything for each `fetch::end::html` (would that make sense? most of the responses are garbage from ads, trackers, etc.) but that will require more time.

Reply:
I'm not talking about doing it for every `html` event, what I mean is why do this in two events, instead of doing the validation directly in the event `fetch::end::html`.